lib/suite.js

```diff
@@ -131,7 +131,7 @@ Suite.prototype.timeout = function(ms) {
  * @return {Suite|number} for chaining
  */
 Suite.prototype.retries = function(n) {
-  if (!arguments.length) {
+  if (!arguments.length || typeof n === 'undefined') {
```

Comment:
this *could* break something. even though it's "private", it's called by `Mocha#retries` which is public.

Reply:
Think this change is incorrect. If we _were_ passed anything then it was meant for the mutator portion of the method. 

In this case, we should verify we were passed a number and `floor` it.
* if it's not the right type, then `TypeError` code=`ERR_INVALID_ARG_TYPE`
* if right type but bad value (e.g., `Number.NaN`), `TypeError` code= `ERR_INVALID_ARG_VALUE`.

Reply:
> Think this change is incorrect. If we were passed anything then it was meant for the mutator portion of the method.

That's what the code does... but I'm not convinced that was the intent.  The API is inconsistent for setter/getter methods across the codebase, which is problematic.  Some methods check for `arguments.length`, others check if the first parameter is `undefined`, and probably other strategies or combinations thereof.

The use case is:

```js
suite.retries(foo.bar)
```

Where Object `foo` *has no property* `bar`.

I'll just hack around this elsewhere.  The API needs to be made consistent, and that means settling on a strategy.  I don't love setters/getters *anyway*, so maybe that strategy is "remove them where possible."

- [ ] revert and fix elsewhere